db/migrate/20240415052153_fixup_upload_reference_created_at.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class FixupUploadReferenceCreatedAt < ActiveRecord::Migration[7.0]
+  def up
+    ActiveRecord::Base.transaction do
+      # Target: Post
+      execute <<~SQL
+        UPDATE upload_references
+        SET created_at = posts.created_at, updated_at = NOW()
```

Comment:
This is still imperfect yes, as the upload could be edited into the post later, but it's still more granular and accurate than what we have now.